File: com/teamiurg/techexpansion/block/TEBlocks.java
package com.teamiurg.techexpansion.block;

import com.teamiurg.techexpansion.TechExpansion;
import com.teamiurg.techexpansion.engine.render.RenderHandler;

import net.minecraft.block.Block;
import net.minecraft.block.material.Material;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraftforge.fml.common.registry.GameRegistry;

public class TEBlocks {
	
	
	/* ores */
	public static Block copper_ore = new TEBlock(Material.iron, "copper_ore", TechExpansion.tab_techExpansion, 3.0f, 15.0f, Block.soundTypeStone, 0, 0.0f, "pickaxe", 1, false);
	public static Block aluminum_ore = new TEBlock(Material.iron, "aluminum_ore", TechExpansion.tab_techExpansion, 3.0f, 15.0f, Block.soundTypeStone, 0, 0.0f, "pickaxe", 1, false);
	public static Block tin_ore = new TEBlock(Material.iron, "tin_ore", TechExpansion.tab_techExpansion, 2.5f, 13.0f, Block.soundTypeStone, 0, 0.0f, "pickaxe", 1, false);
	/* blocks */
	public static Block copper_block = new TEBlock(Material.iron, "copper_block", TechExpansion.tab_techExpansion, 3.5f, 17.0f, Block.soundTypeMetal, 0, 0.0f, "pickaxe", 1, false);
	public static Block aluminum_block = new TEBlock(Material.iron, "aluminum_block", TechExpansion.tab_techExpansion, 3.5f, 17.0f, Block.soundTypeMetal,0 ,0.0f, "pickaxe", 1, false);
	public static Block tin_block = new TEBlock(Material.iron, "tin_block", TechExpansion.tab_techExpansion, 3.0f, 15.0f, Block.soundTypeMetal, 0, 0.0f, "pickaxe", 1, false);
	
	public static void registerBlocks() {
		/* ores */
		GameRegistry.registerBlock(copper_ore, "copper_ore");
		GameRegistry.registerBlock(aluminum_ore, "aluminum_ore");
		GameRegistry.registerBlock(tin_ore, "tin_ore");
		/* blocks */
		GameRegistry.registerBlock(copper_block, "copper_block");
		GameRegistry.registerBlock(aluminum_block, "aluminum_block");
		GameRegistry.registerBlock(tin_block, "tin_block");
	}

	public static void registerRenders() {
		/* ores */
		RenderHandler.registerRender(copper_ore);
		RenderHandler.registerRender(aluminum_ore);
		RenderHandler.registerRender(tin_ore);
		/* blocks */
		RenderHandler.registerRender(copper_block);
		RenderHandler.registerRender(aluminum_block);
		RenderHandler.registerRender(tin_block);
	}
}

File: src/main/scala/com/teamisotope/techexpansion/block/TEBlocks.scala
package com.teamisotope.techexpansion.block

import com.teamisotope.techexpansion.TechExpansion
import com.teamisotope.techexpansion.block.special.{GlassChamber, ItemRawMineral, RawMineral, VacuumPump}
import com.teamisotope.techexpansion.mineral.{Mineral, MineralRegistry}
import com.teamisotope.techexpansion.util.RenderUtils
import com.teamisotope.techexpansion.world.{EnumDimension, Ore}
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.item.ItemBlock
import net.minecraftforge.fml.common.registry.GameRegistry
import net.minecraftforge.oredict.OreDictionary

object TEBlocks {

  val raw_mineral: RawMineral = new RawMineral()
  val raw_mineral_item: ItemRawMineral = new ItemRawMineral()

  val galena: Mineral = new Mineral().setUnlocName("techexpansion.mineral.galena").setRegistryName("techexpansion:galena")
  val native_silver: Mineral = new Mineral().setUnlocName("techexpansion.mineral.native_silver").setRegistryName("techexpansion:native_silver")
  val tetrataenite: Mineral = new Mineral().setUnlocName("techexpansion.mineral.tetrataenite").setRegistryName("techexpansion:tetrataenite")
  val bauxite: Mineral = new Mineral().setUnlocName("techexpansion.mineral.bauxite").setRegistryName("techexpansion:bauxite")
  val cassiterite: Mineral = new Mineral().setUnlocName("techexpansion.mineral.cassiterite").setRegistryName("techexpansion:cassiterite")
  val native_gold: Mineral = new Mineral().setUnlocName("techexpansion.mineral.native_gold").setRegistryName("techexpansion:native_gold")
  val native_copper: Mineral = new Mineral().setUnlocName("techexpansion.mineral.native_copper").setRegistryName("techexpansion:native_copper")

  val copper_block: TEBlock = new TEBlock(Material.ROCK, "copper_block", TechExpansion.tab_blocks, 3.0f, 10.0f, "pickaxe", 1)
  val tin_block: TEBlock = new TEBlock(Material.ROCK, "tin_block", TechExpansion.tab_blocks, 3.0f, 10.0f, "pickaxe", 1)
  val aluminum_block: TEBlock = new TEBlock(Material.ROCK, "aluminum_block", TechExpansion.tab_blocks, 3.0f, 10.0f, "pickaxe", 1)
  val nickel_block: TEBlock = new TEBlock(Material.ROCK, "nickel_block", TechExpansion.tab_blocks, 3.0f, 10.0f, "pickaxe", 1)
  val silver_block: TEBlock = new TEBlock(Material.ROCK, "silver_block", TechExpansion.tab_blocks, 3.0f, 10.0f, "pickaxe", 1)
  val lead_block: TEBlock = new TEBlock(Material.ROCK, "lead_block", TechExpansion.tab_blocks, 3.0f, 10.0f, "pickaxe", 1)

  val glass_chamber: TEBlock = new GlassChamber()
  val vacuum_pump: TEBlock = new VacuumPump()

  def register(): Unit = {
    GameRegistry.register(raw_mineral)
    GameRegistry.register(raw_mineral_item)
    MineralRegistry.registerMineral(galena)
    MineralRegistry.registerMineral(native_silver)
    MineralRegistry.registerMineral(tetrataenite)
    MineralRegistry.registerMineral(bauxite)
    MineralRegistry.registerMineral(cassiterite)
    MineralRegistry.registerMineral(native_gold)
    MineralRegistry.registerMineral(native_copper)

    registerBlock(copper_block)
    OreDictionary.registerOre("blockCopper", copper_block)
    registerBlock(tin_block)
    OreDictionary.registerOre("blockTin", tin_block)
    registerBlock(aluminum_block)
    OreDictionary.registerOre("blockAluminum", aluminum_block)
    registerBlock(nickel_block)
    OreDictionary.registerOre("blockNickel", nickel_block)
    registerBlock(silver_block)
    OreDictionary.registerOre("blockSilver", silver_block)
    registerBlock(lead_block)
    OreDictionary.registerOre("blockLead", lead_block)

    registerBlock(vacuum_pump)
    registerBlock(glass_chamber)
  }

  def renders(): Unit = {



    RenderUtils.register(copper_block, "resources")
    RenderUtils.register(tin_block, "resources")
    RenderUtils.register(aluminum_block, "resources")
    RenderUtils.register(nickel_block, "resources")
    RenderUtils.register(silver_block, "resources")
    RenderUtils.register(lead_block, "resources")

    RenderUtils.register(vacuum_pump, "machines")
    RenderUtils.register(glass_chamber, "machines")
  }

  private def registerBlock(b: TEBlock) {
    GameRegistry.register(b)
    GameRegistry.register(b.itemblock)
  }

}
